File: python/sort/selection.py
# This is Selection sort.
# Worst: O(n^2)
# Best: O(n^2)
# Average: O(n^2)

def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        temp = i
        for j in range(i+1, n):
            if arr[j] < arr[temp]:
                temp = j
        arr[i], arr[temp] = arr[temp], arr[i]

File: rust/sort/selection.rs
// This is Selection sort.
// Worst: O(n^2)
// Best: O(n^2)
// Average: O(n^2)

fn selection<T: Ord>(arr: &mut Vec<T>){
    let n = arr.len()
    for i in 0..n{
        let mut temp = i
        for j in i + 1..n{
            if arr[j] < arr[temp]{
                temp = j
            }
        }
        arr.swap(i, temp)
    }
}